packages/core/strapi/lib/commands/admin-create.js

```diff
@@ -90,7 +91,20 @@ module.exports = async function(cmdOptions = {}) {
 };
 
 async function createAdmin({ email, password, firstname, lastname }) {
-  const app = await strapi().load();
+  const appDir = process.cwd();
+
+  const isTSProject = await tsUtils.isUsingTypeScript(appDir);
+  const outDir = await tsUtils.resolveOutDir(appDir);
+
+  if (isTSProject)
+    await tsUtils.compile(appDir, {
+      watch: false,
+      configOptions: { options: { incremental: true } },
+    });
+
+  const distDir = isTSProject ? outDir : appDir;
```

Comment:
couldn't this be returned in the compile meta data to encapsulate as much as possible in the tsUtils ?